crates/miden-objects/src/account/storage/map.rs

```diff
@@ -109,6 +169,19 @@ impl StorageMap {
 
         self.root()
     }
+
+    /// Consumes the map and returns the underlying map of entries.
+    pub fn into_entries(self) -> BTreeMap<RpoDigest, Word> {
+        self.map
+    }
+
+    /// Hashes the given key to get the key of the SMT.
+    pub fn hash_key(key: RpoDigest) -> RpoDigest {
+        let padded_input =
+            [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::ZERO, key[0], key[1], key[2], key[3]];
```

Comment:
Why does this need padding?

Reply:
If we use the `hash` instruction in MASM - we shouldn't need this - i.e., we should be able to do something like:
```rust
Hasher::hash_elements(key.as_elements())
```

Reply:
Agreed! Let's create an issue for this in the VM.

Reply:
Created https://github.com/0xPolygonMiden/miden-vm/issues/1730 :+1:.